File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // check for 2 arguments, if not print error message and return.
    if (argc != 2)
    {
        printf("Usage: ./caesar k \n");
        return 1;
    }

    // store valid key in int variable
    int k = atoi(argv[1]);

    // prompt user for a message to encrypt
    string message = get_string("plaintext: ");
    printf("ciphertext: ");
    for (int i = 0, n = strlen(message); i < n; i++)
    {
        //shift letters by k
        if islower(message[i])
        {
            printf("%c", (((message[i] + k) - 97) % 26) + 97);
        }
        else if isupper(message[i])
        {
            printf("%c", (((message[i] + k) - 65) % 26) + 65);
        }
        else
            printf("%c", message[i]);
    }
    printf("\n");
    return 0;
}

File: caesar.py
import sys
from cs50 import get_string

if (len(sys.argv) == 2):

    k = int(sys.argv[1])

    if (k >= 0):

        #Prompt plaintext
        message = get_string("plaintext: ")

        #Encrypt plaintext
        print("ciphertext: ", end="")
        for c in message:
            if c.isupper():

                #convert character to ascii code and shift
                a = (((ord(c) - 65 + k) % 26) + 65)

                #convert ascii code of shifted char into char to print
                print(f"{chr(a)}", end="")

            elif c.islower():

                b = (((ord(c) - 97 + k) % 26) + 97)
                print(f"{chr(b)}", end="")

            else:
                print(f"{c}", end="")


        print("")
    else:
        print("Key must be a non-negative integer.")

else:
    print("Usage: ./caesar k")
    exit(1)



